lib/apartment/adapters/abstract_adapter.rb

```diff
@@ -116,12 +126,10 @@ def setup_connection_specification_name
         Apartment.connection_class.connection_specification_name = nil
         Apartment.connection_class.instance_eval do
           def connection_specification_name
-            if !defined?(@connection_specification_name) || @connection_specification_name.nil?
-              apartment_spec_name = Thread.current[:_apartment_connection_specification_name]
-              return apartment_spec_name ||
-                (self == ActiveRecord::Base ? "primary" : superclass.connection_specification_name)
-            end
-            @connection_specification_name
+            return :_apartment_excluded if @connection_specification_name == :_apartment_excluded
```

Comment:
The old code had a path to return `@connection_specification_name` if it existed where this new one doesn't. Was that just not necessary anymore?

Reply:
I ended up removing that path to fix issues with our test suite. The spec name was getting set (and cached in that instance variable) as `primary`, which I believe was then preventing connections to the database specified by the Thread's `_apartment_connection_specification_name` value.

Reply:
It could have been an issue with the the timing of how Apartment was being loaded, but I was unable to see anything obvious.